include/dr/mhp/algorithms/inclusive_exclusive_scan_impl.hpp

```diff
@@ -57,8 +59,32 @@ auto inclusive_exclusive_scan_impl_(R &&r, O &&d_first, BinaryOp &&binary_op,
   using value_type = U;
   assert(aligned(r, d_first));
 
-  bool use_sycl = mhp::use_sycl();
   auto comm = default_comm();
+  bool use_sycl = mhp::use_sycl();
+
+  if (rng::size(r) <= 2 * comm.size()) {
+    std::vector<value_type> vec_in(rng::size(r));
+    std::vector<value_type> vec_out(rng::size(r));
+    mhp::copy(0, r, vec_in.begin());
+
+    if (comm.rank() == 0) {
+      if constexpr (is_exclusive) {
+        assert(init.has_value());
+        std::exclusive_scan(detail::direct_iterator(vec_in.begin()),
+                            detail::direct_iterator(vec_in.end()),
+                            detail::direct_iterator(vec_out.begin()),
+                            init.value(), binary_op);
+      } else {
+        std::inclusive_scan(detail::direct_iterator(vec_in.begin()),
```

Comment:
initt is ignored but it can be non-empty
add 3 test cases for this code: exclusive scan with nonzero init, inclusive scan without init, inclusive scan with init

Reply:
Done